xen/arch/x86/smpboot.c

```diff
@@ -483,8 +483,8 @@ static int wakeup_secondary_cpu(int phys_apicid, unsigned long start_eip)
     if ( tboot_in_measured_env() && !tboot_wake_ap(phys_apicid, start_eip) )
         return 0;
 
-    if ( slaunch_active )
-        return slaunch_wake_aps(start_eip);
+    if ( slaunch_active && boot_cpu_data.x86_vendor == X86_VENDOR_INTEL )
```

Comment:
`ap_boot_method` was created specifically for handling nonstandard SMP wake-ups, I think it should be used here as well. Not sure where would be proper place to set it, it isn't linked with early code, so perhaps somewhere around mapping and protecting TXT memory, unless you have better ideas.

Reply:
Made use of it in d9176ce93703eb8d1c337b3d67e5e1c3d2042ff6.